src/nodeBuffer.ml

```diff
@@ -44,29 +45,341 @@ external allocFillStringWithEncoding: int -> fill:Js.String.t ->
+  "readUint16BE" [@@bs.send]
+let readUint16BENoAssert buffer ~offset = readUint16BENoAssert buffer ~offset ~noAssert:Js.true_
+
+external readUint16LE : t -> offset:int -> float = "" [@@bs.send]
+external readUint16LENoAssert : t -> offset:int -> noAssert:Js.boolean -> float =
+  "readUint16LE" [@@bs.send]
+let readUint16LENoAssert buffer ~offset = readUint16LENoAssert buffer ~offset ~noAssert:Js.true_
+
+external readUint32BE : t -> offset:int -> float = "" [@@bs.send]
+external readUint32BENoAssert : t -> offset:int -> noAssert:Js.boolean -> float =
+  "readUint32BE" [@@bs.send]
+let readUint32BENoAssert buffer ~offset = readUint32BENoAssert buffer ~offset ~noAssert:Js.true_
+
+external readUint32LE : t -> offset:int -> float = "" [@@bs.send]
+external readUint32LENoAssert : t -> offset:int -> noAssert:Js.boolean -> float =
+  "readUint32LE" [@@bs.send]
+let readUint32LENoAssert buffer ~offset = readUint32LENoAssert buffer ~offset ~noAssert:Js.true_
+
+external readUintBE : t -> offset:int -> length:int -> float = "" [@@bs.send]
+external readUintBENoAssert : t -> offset:int -> length:int -> noAssert:Js.boolean ->
+  float = "readUintBE" [@@bs.send]
+let readUintBENoAssert buffer ~offset ~length = readUintBENoAssert buffer ~offset
+  ~length ~noAssert:Js.true_
+
+external readUintLE : t -> offset:int -> length:int -> float = "" [@@bs.send]
+external readUintLENoAssert : t -> offset:int -> length:int -> noAssert:Js.boolean ->
+  float = "readUintLE" [@@bs.send]
+let readUintLENoAssert buffer ~offset ~length = readUintLENoAssert buffer ~offset
+  ~length ~noAssert:Js.true_
+
+external slice : t -> t = "" [@@bs.send]
+external sliceOffset : t -> start:int -> t = "slice" [@@bs.send]
+external sliceRange : t -> start:int -> end_:int -> t = "slice" [@@bs.send]
+
+external swap16 : t -> t = "" [@@bs.send]
+external swap32 : t -> t = "" [@@bs.send]
+external swap64 : t -> t = "" [@@bs.send]
+
+external toJSON : t -> < .. > Js.t = "" [@@bs.send]
+
+external toString: t -> Js.String.t = "" [@@bs.send]
+external toStringWithEncoding: t -> encoding:Js.String.t -> Js.String.t =
+  "toString" [@@bs.send]
+external toStringWithEncodingOffset: t -> encoding:Js.String.t -> start:int
+  -> Js.String.t = "toString" [@@bs.send]
+external toStringWithEncodingRange: t -> encoding:Js.String.t -> start:int
+  -> end_:int -> Js.String.t = "toString" [@@bs.send]
+
+(* FIXME after iterators support *)
+(* external values : t -> Iterator = "" [@@bs.get] *)
+
+external write : t -> Js.String.t -> int = "" [@@bs.send]
+external writeOffset : t -> value:Js.String.t -> offset:int -> int =
+  "write" [@@bs.send]
+external writeRange : t -> value:Js.String.t -> offset:int -> length:int -> int =
+  "write" [@@bs.send]
+external writeRangeWithEncoding : t -> value:Js.String.t -> offset:int ->
+  length:int -> encoding:Js.String.t -> int = "write" [@@bs.send]
+
+external writeDoubleBE : t -> value:float -> offset:int -> float = "" [@@bs.send]
```

Comment:
I tried to stay close to original names as much as possible, but okay, names are long anyway, no harm with a little bit more explicitness.